### PR TITLE
feat(examples): add cra

### PR DESCRIPTION
**Pull request by tknickman**
Create react app example.

### CODE REVIEW

File: examples/with-create-react-app/apps/docs/.gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# dependencies
+/node_modules
```
Comment (jaredpalmer) [inline examples/with-create-react-app/apps/docs/.gitignore:4]: Can we have one gitignore at the root?

File: examples/with-create-react-app/apps/web/package.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    "test": "react-scripts test",
+    "eject": "react-scripts eject",
+    "lint": "TIMING=1 eslint src --fix",
+    "clean": "rm -rf build"
+  },
+  "dependencies": {
+    "ui": "*",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-router-dom": "^6.2.1",
```
Comment (jaredpalmer) [inline examples/with-create-react-app/apps/web/package.json:18]: Do we need react-router?

File: examples/with-create-react-app/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+- `tsconfig`: `tsconfig.json`s used throughout the monorepo
+
+Each package/app is 100% [TypeScript](https://www.typescriptlang.org/).
+
+### Utilities
+
+This Turborepo has some additional tools already setup for you:
+
+- [TypeScript](https://www.typescriptlang.org/) for static type checking
+- [TypeScript](https://www.typescriptlang.org/) for static type checking
```
[the remark this answers is not in the document]
Comment (tknickman) [reply examples/with-create-react-app/README.md:24]: 👍🏼 good catch.

File: examples/with-create-react-app/apps/web/package.json (hunk 2 above)
Comment (tknickman) [reply examples/with-create-react-app/apps/web/package.json:18]: Yea interestingly this ships with CRA, but it's not used in the example page so we can definitely yank it.

File: examples/with-create-react-app/apps/docs/package.json [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    "@testing-library/react": "^12.1.2",
+    "@testing-library/user-event": "^13.5.0",
+    "@types/jest": "^27.0.3",
+    "@types/node": "^16.11.14",
+    "@types/react": "^17.0.37",
+    "@types/react-dom": "^17.0.11",
+    "tsconfig": "*",
+    "typescript": "^4.5.4"
+  },
+  "eslintConfig": {
```
[the remark this answers is not in the document]
Comment (tknickman) [reply examples/with-create-react-app/apps/docs/package.json:33]: This is included in the vanilla CRA install and how they configure linting. 

I messed around with abstracting this out briefly but I didn't want to eject, and ran into a few issues with getting it to lint cleanly. But let me revisit that quickly before wrapping this one.

File: examples/with-create-react-app/packages/ui/package.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  "private": true,
+  "main": "./dist/index.js",
+  "module": "./dist/index.mjs",
+  "types": "./dist/index.d.ts",
+  "sideEffects": false,
+  "license": "MIT",
+  "scripts": {
+    "build": "tsup src/index.tsx --format esm,cjs --dts --external react",
+    "dev": "tsup src/index.tsx --format esm,cjs --watch --dts --external react",
+    "lint": "eslint src/*.ts*",
```
Comment (nathanhammond) [inline examples/with-create-react-app/packages/ui/package.json:13]: Missing an `eslint` dependency in this file.

File: examples/with-create-react-app/apps/web/package.json (hunk 2 above)
Comment (nathanhammond) [reply examples/with-create-react-app/apps/web/package.json]: I'm generally opposed to changing much that comes out-of-the-box with contents of a generator—even if unused. We want the upstream documentation about how to do things to still apply. Dropping this just trolls somebody who is getting started with CRA and doesn't know why their thing doesn't work.

File: examples/with-create-react-app/apps/docs/.gitignore (hunk 1 above)
Comment (nathanhammond) [reply examples/with-create-react-app/apps/docs/.gitignore]: This also falls under my, "stick with the generator" opinion. Nested `.gitignore` is valid, fine, and clear. 🤷‍♂️

**Review by nathanhammond — APPROVED** [review]
Only one thing included in the fixship: the missing eslint dep.

File: examples/with-create-react-app/apps/web/package.json (hunk 2 above)
Comment (tknickman) [reply examples/with-create-react-app/apps/web/package.json]: That's a fair point. Generally speaking I did my best to leave things that started from a generator alone for this exact reason - we don't want to have to support "turbo's CRA". So I would be fine putting this back, but it's also a pretty easy error to see and fix if someone does run into this down the line.

**Pull request merged**